### PR TITLE
Kh/#58/authorization poro

### LINKED ISSUE #58: Authorization PORO

### PR DESCRIPTION
**Pull request by kheppenstall**
Closes #58 

@Dpalazzari @case-eee @jdconrad89 @Robbie-Smith This PR adds an authorization PORO. We have a lot of edge cases we neglected testing around roles so should take another look at our feature tests to make sure we are testing for authorization.

Let's get at least 2 reviews of this one. Adding authorization made a lot of tests fail so I touched a lot of files to fix things.

### CODE REVIEW

File: app/controllers/api/v1/project_owners_controller.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-class Api::V1::ProjectOwnersController < ApplicationController
+class Api::V1::ProjectOwnersController < Api::V1::BaseController
   def index
```
Comment (Dpalazzari) [inline app/controllers/api/v1/project_owners_controller.rb]: @kheppenstall Why did we do this?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -2,8 +2,20 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
   helper_method :current_user
+  add_flash_types :success, :info, :warning, :danger
```
Comment (Dpalazzari) [inline app/controllers/application_controller.rb]: 👍

File: app/models/project.rb [hunk 3]
```diff
@@ -37,9 +38,10 @@ def owner_names
     owners.pluck(:name).to_sentence
   end
 
-  def self.create_project(attributes)
-    project       = Project.new(attributes)
-    project.slug  = project.generate_slug
+  def self.create_project(attributes, owner)
+    project       = owner.owned_projects.new(attributes)
+    project.generate_slug
+    project.owners.each {|owner| owner.add_owner_role }
```
Comment (Dpalazzari) [inline app/models/project.rb]: I will change this to just `current_user.add_owner_role` and move it to the controller in my branch.

File: app/controllers/api/v1/project_owners_controller.rb (hunk 1 above)
Comment (Dpalazzari) [inline app/controllers/api/v1/project_owners_controller.rb]: nvm, @drod1000 filled me in.

File: spec/support/feature_helpers.rb [hunk 4]
```diff
@@ -5,6 +5,13 @@ def logged_in_user
     user
   end
 
+  def logged_in_admin
+    admin = create(:user)
+    admin.roles << create(:role, name: 'admin')
+    page.set_rack_session(user_id: admin.id)
+    admin
+  end
```
Comment (drod1000) [inline spec/support/feature_helpers.rb]: 👍

File: app/models/project.rb (hunk 3 above)
Comment (kheppenstall) [inline app/models/project.rb]: 👍
